iravelmakina/caesar-encryption-algorithm-ivelmakina
# Key normalization in Caesar Cipher

### Description:
Fixed the encrypt/decrypt functions by adding key normalization to ensure the key is within the valid range (0-25), preventing incorrect encryption results for large key values. Rebuilt all files to work correctly with the updated logic and adjusted project configurations accordingly.

### Changes:
- Added key normalization with `key = key % 26` in the encryption/decryption logic.
- Updated the Caesar cipher implementation to handle large key values correctly.
- Rebuilt every file to work correctly with the updated logic.
- Adjusted configurations to reflect these changes.

### How to Test:
1. Compile `caesar.cpp` and `main.cpp` to create `caesar.dll` and `main.exe`
2. Compile `caesar.cpp` and `main.cpp` to create `caesar.lib` and `main.exe`
3. Compile `program_cli.cpp` to create `program_cli.exe`
4. Run the test program to ensure it works as expected.
5. Verify the output is correct for various key values, including large values like 300.

### Screenshots:
1-3. *See [Caesar Encryption Algorithm](https://github.com/iravelmakina/caesar-encryption-algorithm-ivelmakina/pull/1)*
4-5. 
![image](https://github.com/iravelmakina/caesar-encryption-algorithm-ivelmakina/assets/150923968/5c22077e-27a3-45a8-86ac-4784ea8a227e) 
![image](https://github.com/iravelmakina/caesar-encryption-algorithm-ivelmakina/assets/150923968/c498bfa3-2a27-4be4-81ef-a1726b0e41d6)
![image](https://github.com/iravelmakina/caesar-encryption-algorithm-ivelmakina/assets/150923968/d540f11b-9f93-4873-bfa1-c8cd1d465ad9)
